d3a_interface/settings_validators.py

```diff
@@ -15,75 +15,77 @@
-    if "cloud_coverage" in settings_dict and \
+            raise D3ASettingsException(f"Invalid slot_length "
+                                       f"({slot_length.in_minutes()} min, limits: "
+                                       f"[{min_slot_length.in_minutes()} min, "
+                                       f"{max_slot_length.in_minutes()} min])")
+    if ("cloud_coverage" in settings_dict and
         not (ConstSettings.PVSettings.CLOUD_COVERAGE_LIMIT[0]
              <= settings_dict["cloud_coverage"] <=
-             ConstSettings.PVSettings.CLOUD_COVERAGE_LIMIT[1]):
-        raise D3ASettingsException(f'Invalid cloud coverage value '
-                                   f'({settings_dict["cloud_coverage"]}).')
-    if "spot_market_type" in settings_dict and \
-            not ConstSettings.IAASettings.MARKET_TYPE_LIMIT[0] \
-            <= settings_dict["spot_market_type"] <= \
-            ConstSettings.IAASettings.MARKET_TYPE_LIMIT[1]:
-        raise D3ASettingsException(f'Invalid value ({settings_dict["spot_market_type"]}) '
-                                   f'for spot market type.')
+             ConstSettings.PVSettings.CLOUD_COVERAGE_LIMIT[1])):
+        raise D3ASettingsException(f"Invalid cloud coverage value "
+                                   f"({settings_dict['cloud_coverage']}).")
+    if ("spot_market_type" in settings_dict and
+            not ConstSettings.IAASettings.MARKET_TYPE_LIMIT[0]
+            <= settings_dict["spot_market_type"] <=
+            ConstSettings.IAASettings.MARKET_TYPE_LIMIT[1]):
+        raise D3ASettingsException(f"Invalid value ({settings_dict['spot_market_type']}) "
+                                   f"for spot market type.")
     if "sim_duration" in settings_dict and not slot_length <= settings_dict["sim_duration"]:
         raise D3ASettingsException(f"Invalid simulation duration "
                                    f"(lower than slot length of {slot_length.minutes} min")
     if "market_count" in settings_dict and not 1 <= settings_dict["market_count"]:
         raise D3ASettingsException("Market count must be greater than 0.")
-    if "max_panel_power_W" in settings_dict and not \
-            ConstSettings.PVSettings.MAX_PANEL_OUTPUT_W_LIMIT[0] \
-            <= settings_dict["max_panel_power_W"] \
-            <= ConstSettings.PVSettings.MAX_PANEL_OUTPUT_W_LIMIT[1]:
-        raise D3ASettingsException(f'Invalid value for max_panel_power_W '
-                                   f'({settings_dict["max_panel_power_W"]}).')
-    if "grid_fee_type" in settings_dict and \
-            int(settings_dict["grid_fee_type"]) not in ConstSettings.IAASettings.VALID_FEE_TYPES:
-        raise D3ASettingsException(f'Invalid value for grid_fee_type '
-                                   f'({settings_dict["grid_fee_type"]}).')
+    if ("capacity_kW" in settings_dict and not
+            ConstSettings.PVSettings.CAPACITY_KW_LIMIT[0]
```

Comment:
`RangeLimit` has `min` and `max` attributes for this purpose. They would help to make the code clearer:
```suggestion
            ConstSettings.PVSettings.CAPACITY_KW_LIMIT.min
```

Reply:
The same applies for `CAPACITY_KW_LIMIT[1]`.